R/helper_testing.R

```diff
@@ -60,10 +60,66 @@ prepare_example_db <- function(dir, add_data = FALSE, add_job_table = FALSE,
     DBI::dbExecute(con, query_text)
   }
 
+  if (add_fk_data) {
+    add_fk_data(con)
+  }
+
   DBI::dbDisconnect(con)
   path
 }
 
+#' Add tables with foreign key constraints for testing
+#'
+#' Create some tables with foreign key constraints and add some simple data
+#' to them for testing.
+#'
+#' @param con Connection to DB to add tables with foreign key constraints
+#'
+#' @keywords internal
+add_fk_data <- function(con) {
+  DBI::dbExecute(con,
+    "CREATE TABLE region (
+      name TEXT PRIMARY KEY,
+      parent TEXT,
+      FOREIGN KEY (parent) REFERENCES region(name)
+    )")
+    region1 <- data.frame(list(
```

Comment:
what is the `list()` call doing here? (& other `data.frame(list(...))` constructions throughout)

Reply:
Not necessary again, I've removed those too.